lib/plugin_helper.rb

```diff
@@ -0,0 +1,65 @@
+module RubyPluginHelper
+  # 'template' is the value of `target_mapping` from the inventory
+  # Returns an array of strings which are the lookups from the template
+  def config_lookup(template)
+    lookups = []
+    postwalk_vals(template) do |value|
+      # First collect the lookups to make
+      lookups << value if value.is_a?(String)
+      value
+    end
+    lookups.uniq.map { |lookup| lookup.split('.') }
+  end
+
+  # 'template' is the value of `target_mapping` from the inventory
+  # 'lookups' is an array of hashes, where each hash is the lookup for a target
+  # Returns an array of hashes to be sent to the inventory
+  def populate_mapping(template, lookups)
```

Comment:
I think this function is simpler and more versatile if it just accepts a single data object and template and any looping is left up to the caller.

bikeshed -> `apply_mapping(mapping, data)`?